corehq/apps/case_importer/do_import.py

```diff
@@ -119,22 +126,40 @@ def commit_caseblocks(self):
     def submit_caseblocks(self, caseblocks):
         if not caseblocks:
             return
+        if rate_limit_submission(self.domain):
+            # the duration of the last submission is a combined heuristic
+            # for the amount of load on the databases
+            # and the amount of load that the requests from this import put on the databases.
+            # The amount of time to wait, during a high submission period
+            # and while this project is using up more than its fair share
+            # should be proportional to this heuristic.
+            # For a fully throttled domain, this will up to double
+            # the amount of time the case import takes
+            datadog_counter(
+                'commcare.case_importer.import_delayed_duration',
+                self._last_submission_duration, tags=[
+                    'domain:{}'.format(self.domain)
```

Comment:
Would it make sense to use a bucket counter here so we can see an approximate distribution of delay duration? I believe adding that would simply entail adding a `duration:<value_bucket>` tag.

Related: did you intend to pass `self._last_submission_duration` as the counter increment value? (It's not always an integer value, is it?)

Reply:
I didn't realize count only accepted integer values. I was intending to use this to make a bar graph where the height of the bar is the sum of the lengths of the delays during that period, so we I can validate that the throttling is achieving the goals of (1) being more aggressive when server load is high (2) staying completely out of the way when server usage is low.

But I can't find any other examples of us doing this with float values, so I'll assume you're right and this doesn't work the way I want it to. I'll change this to counting the number of delays and bucketing by duration like you suggested.